lib/semantics/canonicalize-do.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "canonicalize-do.h"
+#include "../parser/parse-tree-visitor.h"
+
+namespace Fortran::semantics {
```

Comment:
A lot of this code might be made more brief if it were in `namespace Fortran::parser`.  It doesn't seem to depend on much from semantics.

Reply:
done